pkg/services/cpd/cpd.go

```diff
@@ -64,3 +114,93 @@ func InvestigateTriggered(r *investigation.Resources) error {
 	// we would silence the alert when we had a service log in the case of network verifier detecting failures.
 	return r.PdClient.EscalateAlert()
 }
+
+func isSubnetRouteValid(awsClient aws.Client, subnetID string) (bool, error) {
+	var routeTable string
+
+	// Try and find a Route Table associated with the given subnet
+	describeRouteTablesOutput, err := awsClient.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   awsSdk.String("association.subnet-id"),
+				Values: []*string{awsSdk.String(subnetID)},
+			},
+		},
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to describe route tables associated to subnet %s: %w", subnetID, err)
+	}
```

Comment:
**nitpick:**
We could wrap this into a function reflecting the above comment. This would make the code self explanatory and the function could move into the `aws.go` file. 
```
getSubnetRouteTables(subnetID string) ... {}
```